_docs/treeview/treeview.md

```diff
@@ -0,0 +1,131 @@
+| loadMethod | function | null | no | The method used to load the children of a node when loadOnDemand is true.<br>`function(item: TreeItem)`<br>returns a promise or a list of Objects |
+| filter | function | default function | no | The default function searches according to the node title with exact prefix matching.<br>`function(item: Object)`|
+| onSelect | function | null | no | Triggered when a selection has been changed. |
+| onExpand | function | null | no | Triggered when a node is expanded. |
+| onCollapse | function | null | no | Triggered when a node is collapsed. |
+
+### Methods
+
+| name        | parameters                                  | description               |
+| ----------- | ------------------------------------- | ------------------------------- |
+| collapse | item: Object (required) | Collapses all nodes under a provided node. |
+| collapseAll | none | Collapses all the tree nodes. |
+| expand | item: TreeItem (required) | Expands all nodes under a provided node. |
+| expandAll | Boolean: deep (default) or shallow | Expands all the tree nodes. |
+| selectItem | item: Object (required) | Selects an item in the tree. |
+
+## Input Handling
+
+### Keyboard Navigation
+
+| key | action |
+| --- | --- |
+Home | highlights the first item in the TreeView
+End | highlights the last item in the TreeView
+Enter | Selects current highlighted item (does not expand it though)
+Up arrow | highlights previous item
+Down arrow | highlights next item
+Left arrow | * if child then highlights parent<br>* if parent then collapses it
+Right arrow | * expands item and highlights it<br>* if item was already expanded then highlights the first child
+
+### Code Examples
+
+**Sample data for the examples.**
+
+```
+const treeData: TreeItemData[] = [
+    { label: 'Food Menu', children: [
+        { label: 'Salads', children: [
+            { label: 'Greek Salad' },
+            { label: 'Israeli Salad' },
+            { label: 'Caesar Salad' }
+        ]},
+        { label: 'Steaks', children: [
+            { label: 'Fillet Steak' },
+            { label: 'Sirloin Steak' }
+        ]},
+        { label: 'Desserts', children: [
+            { label: 'Pancakes' },
+            { label: 'Muffin' },
+            { label: 'Waffle' },
+            { label: 'Cupcake' }
+        ]}
+    ]}
+];
+```
+
+**Example views**
+
+* Renders a TreeView
+
+```
+<TreeView dataSource={data} />
+```
+
+* Renders a TreeView with the ability to select an item.
+
+```
+const parentState = {
+  selectedItem: Object;
+};
+
+function onSelectItem(selectedItem) {
+  this.state.selectedItem = selectedItem;
+}
+
+<TreeView dataSource={data} selectedItem={this.state.selectedItem} onSelectItem={this.onSelectItem.bind(this)} />
+```
+
+* Render a TreeView (one icon, several nodes hierarchy) using the traverse children and a non-default item renderer that has a type for a parent node and different type for all children.
```

Comment:
The examples are really meant to drive the development stages. Real-world examples are still to be done, so I left them in the doc in order to provide examples.

I took the above offending example out as it wasn't relevant (in addition to being completely ununderstandable.